packages/stylelint-plugin/src/util/kaizenTokens.ts

```diff
@@ -129,7 +131,74 @@ export const kaizenTokensByName: Readonly<
         moduleName: module.moduleName,
         sassModulePath: module.sassModulePath,
         lessModulePath: module.lessModulePath,
+        color: normalisedColor ? normalisedColor : undefined,
       },
     }
   })
 ).reduce((acc, next) => ({ ...acc, ...next }), {})
+
+/**
+ * This is a record of Kaizen CSS variable values -> Kaizen token SCSS variable names.
+ * It allows you to find the variable for a value (the inverse of getting the value of a variable).
+ * Because colors can be represented in multiple ways, additional keys have been added to here for every variable that holds a color, by running something along the lines of `colorString.to.hex(colorString.get(value).value)`.
+ * What this means is, if a value like $color-white holds (a CSS fallback) value of $fff, you will get the additional color of $ffffff, which is what colorString spits out.
+ * If it is a color, we also add an extra key to the value object (the KaizenToken): `color: ...` so that they are easily identifiable. You might've had to run colorString.get again otherwise.
+ * It looks something like:
+ * ```ts
+ * {
+ *    "#003157": [{name: "$kz-var-color-wisteria-800", color: "#003157", ...}],
+ *    "#fff": [{name: "$kz-var-color-white", color: "#ffffff", value: var(--kz-var-color-white, $fff),...}],
+ *    "#ffffff": [{name: "$kz-var-color-white", color: "#ffffff", ...}],
+ *    "230, 240, 247": [{name: "$kz-var-color-cluny-100-rgb-params", color: undefined, ...}],
+ *    "0 0 12px rgba(0, 0, 0, 0.19)": [{name: "$kz-var-shadow-large-box-shadow", color: undefined, ...}],
+ *    "400": [{name: "$kz-var-typography-paragraph-intro-lede-font-weight", ...}, {name: "$kz-var-typography-paragraph-body-font-weight", ...}, ...],
+ *     "1.5rem": [{name: "$kz-var-spacing-md", ...}]
+ * }
+ * ```
+ */
+export const kaizenTokensByValue: Record<
```

Comment:
I'm questioning the value of this export... given it's not the right shape for us to use as-is in the new rule, is there any point for this record existing in this shape?

Why not just calculate `colorMap` inline in the `prefer-color-tokens.ts` file, based on kaizenTokensByName.

Reply:
You're right the value is questionable.
It's because at some point i decided to 'separate concerns' and this is what came out of it.
Also, though, this map allows us to make a more powerful rule that can recommend things like `1.5rem` be replaced with `$spacing-md`.
I think we should keep it.

Reply:
I'm okay with that. 👍